cli/internal/globby/globby_test.go

```diff
@@ -0,0 +1,520 @@
+			files: []string{
+				"/repos/spanish-inquisition/index.html",
+				"/repos/some-app/dist/index.html",
+				"/repos/some-app/dist/js/index.js",
+				"/repos/some-app/dist/js/lib.js",
+				"/repos/some-app/dist/js/node_modules/browserify.js",
+			},
+			args: args{
+				basePath:        "/repos/some-app/",
+				includePatterns: []string{"**/../../spanish-inquisition/**"},
+				excludePatterns: []string{},
+			},
+			want: []string{},
+		},
+		{
+			name: "traversal works within base path",
+			files: []string{
+				"/repos/some-app/dist/index.html",
+				"/repos/some-app/dist/js/index.js",
+				"/repos/some-app/dist/js/lib.js",
+				"/repos/some-app/dist/js/node_modules/browserify.js",
+			},
+			args: args{
+				basePath:        "/repos/some-app/",
+				includePatterns: []string{"dist/js/../**"},
+				excludePatterns: []string{},
+			},
+			want: []string{
+				"/repos/some-app/dist/index.html",
+				"/repos/some-app/dist/js/index.js",
+				"/repos/some-app/dist/js/lib.js",
+				"/repos/some-app/dist/js/node_modules/browserify.js",
+			},
+		},
+		{
+			name: "self-references (.) work",
+			files: []string{
+				"/repos/some-app/dist/index.html",
+				"/repos/some-app/dist/js/index.js",
+				"/repos/some-app/dist/js/lib.js",
+				"/repos/some-app/dist/js/node_modules/browserify.js",
+			},
+			args: args{
+				basePath:        "/repos/some-app/",
+				includePatterns: []string{"dist/./././**"},
+				excludePatterns: []string{},
+			},
+			want: []string{
+				"/repos/some-app/dist/index.html",
+				"/repos/some-app/dist/js/index.js",
+				"/repos/some-app/dist/js/lib.js",
+				"/repos/some-app/dist/js/node_modules/browserify.js",
+			},
+		},
+		{
+			name: "depth of 1 includes does not capture folders",
+			files: []string{
+				"/repos/some-app/package.json",
+				"/repos/some-app/dist/index.html",
+				"/repos/some-app/dist/js/index.js",
+				"/repos/some-app/dist/js/lib.js",
+				"/repos/some-app/dist/js/node_modules/browserify.js",
+			},
+			args: args{
+				basePath:        "/repos/some-app/",
+				includePatterns: []string{"*"},
+				excludePatterns: []string{},
+			},
+			want: []string{"/repos/some-app/package.json"},
+		},
+		{
+			name: "depth of 1 excludes prevents capturing folders",
+			files: []string{
+				"/repos/some-app/package.json",
+				"/repos/some-app/dist/index.html",
+				"/repos/some-app/dist/js/index.js",
+				"/repos/some-app/dist/js/lib.js",
+				"/repos/some-app/dist/js/node_modules/browserify.js",
+			},
+			args: args{
+				basePath:        "/repos/some-app/",
+				includePatterns: []string{"**"},
+				excludePatterns: []string{"dist/*"},
+			},
+			want: []string{"/repos/some-app/package.json"},
+		},
+	}
+	for _, tt := range tests {
+		t.Cleanup(setup(tt.files))
+		t.Run(tt.name, func(t *testing.T) {
+			got := GlobFiles(tt.args.basePath, tt.args.includePatterns, tt.args.excludePatterns)
+
+			// If the length of both are zero, we're already good to go.
+			if len(got) != 0 || len(tt.want) != 0 {
```

Comment:
`reflect.DeepEqual` with two empty arrays does not match. This special-cases two empty arrays as a test pass.

Is there a better alternative?

Reply:
Hmm, maybe something else is going on? My naive attempt to repro shows them matching: https://go.dev/play/p/ashY45KPrOs?v=goprev

Reply:
One thing to check is maybe `nil` vs `[]string{}`?

Reply:
Fixing the Windows slashes guaranteed I would get an array there so that check is no longer necessary. I've simply removed it.

You called it, though: the `zero-initialized` state of `var result []string` is `nil`.

https://go.dev/play/p/ZG8BX-X9MA-